src/mscorlib/src/System/Text/Encoding.cs

```diff
@@ -978,10 +1003,44 @@ public virtual byte[] GetBytes(String s)
             return bytes;
         }
 
+        // Returns a byte array containing the encoded representation of the given
+        // string range.
+        //
+        [Pure]
+        public unsafe virtual byte[] GetBytes(string s, int index, int count)
+        {
+            if (s == null)
+                throw new ArgumentNullException(nameof(s),
+                    Environment.GetResourceString("ArgumentNull_String"));
+            if (index < 0)
+                throw new ArgumentOutOfRangeException(nameof(index),
+                      Environment.GetResourceString("ArgumentOutOfRange_NeedNonNegNum"));
+            if (count < 0)
+                throw new ArgumentOutOfRangeException(nameof(count),
+                      Environment.GetResourceString("ArgumentOutOfRange_NeedNonNegNum"));
+            if (index > s.Length - count)
+                throw new ArgumentOutOfRangeException(nameof(index),
+                      Environment.GetResourceString("ArgumentOutOfRange_IndexCount"));
+            Contract.EndContractBlock();
+
+            fixed (char* pChar = s)
+            {
+                int byteCount = GetByteCount(pChar + index, count);
+
+                byte[] bytes = new byte[byteCount];
+                fixed (byte* pBytes = &bytes[0])
```

Comment:
This does not sound right. `fixed (byte* pBytes = bytes)` is used in many places.